src/resmom/linux/alps.c

```diff
@@ -1442,12 +1532,17 @@ node_array_start(ud_t *d, const XML_Char *el, const XML_Char **atts)
  * 	This function is registered to handle the node element within an
  * 	inventory response.
  *
- * The standard Expat start handler function prototype is used.
+ * Due to the new XML format introduced in BASIL 1.3, and 
+ * continuing in BASIL 1.4, the count for different arrays need to be
+ * reset at different times.
+ * For example, since BASIL 1.1 and BASIL 1.2 have segments but no sockets
+ * we need to reset the segment_array as part of node_start.  The
+ * segment_array count will be reset in socket_start for BASIL 1.3 and 1.4
```

Comment:
we need to reset the segment_array as part of node_start.  The --> we need to reset the segment_array as part of node_start, however in BASIL 1.3 and 1.4, segment_array count will be reset in socket_start.